server/core/src/config.rs

```diff
@@ -20,24 +20,29 @@ use url::Url;
 
 use crate::repl::config::ReplicationConfiguration;
 
-// Allowed as the large enum is only short lived at startup to the true config
-#[allow(clippy::large_enum_variant)]
-// These structures allow us to move to version tagging of the configuration structure.
 #[derive(Debug, Deserialize)]
 #[serde(untagged)]
+enum VersionDetection {
+    Version(Version),
+    Legacy,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(tag = "version")]
+pub enum Version {
+    #[serde(rename = "2")]
+    V2,
+}
+
+// Allowed as the large enum is only short lived at startup to the true config
+#[allow(clippy::large_enum_variant)]
```

Comment:
maybe replace with `#[expect(clippy::large_enum_variant, reason = "Allowed as the large enum is only short lived at startup to the true config")]`.
So you get a warning when it is no longer met and that it can be removed.

Reply:
I doubt we'll ever be able to remove it until box matching is a thing, so I don't see the value here :(